server/models/getSpots.js

```diff
@@ -20,7 +20,85 @@ const getSpots = () => {
     })
     .catch ((err) => {
       console.log('ERRRO GETTING SPOTS FROM DB IN MODELS', err)
+      throw err;
     })
 };
 
-module.exports.getSpots = getSpots;
\ No newline at end of file
+const getNearbySpots = (lat, long) => {
+  let location = `POINT(${lat} ${long})`;
+  let query = `SELECT spot_id FROM spots WHERE ST_DWithin(geom, ST_GeomFromText($1, 4326)::geography, 500);`;
+  let values = [location];
+  console.log('VALUES', values);
+  return client.query(query, values)
+    .then((spots) => {
+      console.log('NEARBY SPOTS', spots.rows)
+      return spots.rows;
+    })
+    .catch ((err) => {
+      console.log('ERROR GETTING NEARBY SPOTS FROM DB IN MODELS', err);
+      throw err;
+    })
+}
+
+const getFreeSpots = (lat, long, userStartTime, userEndTime) => {
+  let location = `POINT(${lat} ${long})`;
+  let nearbyValues = [location];
+  let nearbyQuery = `SELECT spot_id, host_id, lat, long, address, type, price, photo_url FROM spots WHERE ST_DWithin(geom, ST_GeomFromText($1, 4326)::geography, 1000);`;
+
+  let availableValues = [userStartTime, userEndTime]
+  let availableQuery = `
+    SELECT spot_id
+    FROM bookings
+    WHERE (
+      ($1 >= start_time AND $1 < end_time)
+      OR ($1 < start_time AND $2 > end_time)
+      OR ($2 > start_time AND $2 <= end_time)
+    );
+    `;
+  let promises = [];
+  promises.push(client.query(nearbyQuery, nearbyValues));
+  promises.push(client.query(availableQuery, availableValues));
+
+  return Promise.all(promises)
+    .then((resolved) => {
+      let nearbySpots = resolved[0].rows;
+      let conflictedSpots = resolved[1].rows;
+      console.log('NEARBY', nearbySpots, 'CONFLICTED SPACES', conflictedSpots)
+      let filteredSpots = nearbySpots.filter((nearbySpot) => {
+        for (let i = 0; i < conflictedSpots.length; i++) {
+          if (nearbySpot.spot_id === conflictedSpots[i].spot_id) {
+            return false;
+          }
+          continue;
+        }
+        return true;
+      });
+      console.log('FILTERED SPOTS', filteredSpots)
+      let requiredSpots = filteredSpots.map(({ spot_id, host_id, address, lat, long, type, price, photo_url }) => {
```

Comment:
you could do this in a way that's much shorter but no big deal

Reply:
@matthewwrobel: Can you show me how to do that?

Reply:
Just set a new property on the obj and return it. So instead of destructuring everything you could just do something inside the map callback function body like:

obj.location = {
  lat:1234, 
  log:1234
};
return obj;

I think that would work. Make sense?

Reply:
oh yeah you need to access lat and lng, too. yeah i see why you did it this way because you'd have to delete those unneeded properties.

Reply:
@matthewwrobel: Do you mean something like this: 
let requiredSpots = filteredSpots.map((spot) => {
        return {
          ...spot,
          location: {
            lat: Number(spot.lat),
            lng: Number(spot.long)
          }
        }
      })

Reply:
@matthewwrobel : Yes, that's why I wrote it like that.

Reply:
You don't actually need the spread syntax at all. It'll work if you just assign a new property to each obj and return it. But then you have to delete the old properties you don't need anymore. I get why you did it the way you did.